uniffi_bindgen/src/interface/types/mod.rs

```diff
@@ -34,6 +34,26 @@ pub(super) use finder::TypeFinder;
 mod resolver;
 pub(super) use resolver::{resolve_builtin_type, TypeResolver};
 
+#[derive(Debug, Copy, Clone, Eq, PartialEq, Checksum, Ord, PartialOrd)]
+pub enum ObjectImpl {
+    Struct,
+    Trait,
+}
+
+impl ObjectImpl {
+    /// Return the fully qualified name which should be used by Rust code for
+    /// an object with the given name.
+    /// Includes `r#`, traits get a leading `dyn`. If we ever supported associated types, then
+    /// this would also include them.
+    pub fn rust_name_for(&self, name: &str) -> String {
```

Comment:
This seems oddly named and vaguely mis-placed, but all other names and places seem worse :)